sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -350,4 +350,18 @@ class AnalysisSuite extends AnalysisTest {
 
     assertAnalysisSuccess(query)
   }
+
+  test("SPARK-13651: generator outputs shouldn't be resolved from its child's output") {
+    val input = LocalRelation('key.string, 'value.string)
+    val generatorInput = 'a.map(StringType, IntegerType)
+
+    val query =
+      input
+        .generate(Explode(generatorInput), join = true,
```

Comment:
Is this a valid plan? The input to `Explode` is an `Attribute` named `a`, which is not in the output of `input`.